### ISSUE TITLE
ToolPathResolver broken due to .NET 8.0 SDK Team

### ISSUE THREAD

**Issue by ChristophLindemann** [nuke-build/nuke#1386]
### Usage Information

8.0

### Description

Because the .NET SDK team did some [refactoring in .NET SDK 8.0](https://github.com/dotnet/sdk/pull/33835), the ToolPathResolver is now unable to resolve NuGet packages that are references with `<PackageDownload>` if you also run `dotnet tool restore` (https://github.com/dotnet/sdk/issues/40528).

During `dotnet tool restore` or `dotnet tool install` duplicate package files, which do not include version, are created in the package cache.
So when doing `dotnet tool restore` you will end up with 2 .nupkg files under `%HOME%\.nuget\packages\docfx\2.76.0`
- docfx.2.76.0.nupkg
- docfx.nupkg

Now NuGetPackageResolver will get confused, because it only expects to find exactly 1 file. Since it finds multiple files, it will fail (`SingleOrDefault`)
https://github.com/nuke-build/nuke/blob/6eb779618e22b0cdf1fa27538b305f750b698d88/source/Nuke.Tooling/NuGetPackageResolver.cs#L219

My suggested fix is to use `FirstOrDefault`.

Change
```
candidatePackages.SingleOrDefault(x => x.Version == versionRange.FindBestMatch(candidatePackages.Select(y => y.Version)));
```
to  
```
candidatePackages.FirstOrDefault(x => x.Version == versionRange.FindBestMatch(candidatePackages.Select(y => y.Version)));
```

### Reproduction Steps

1. Run `dotnet tool install docfx -g` command.
2. Check installed files (e.g. `%USERPROFILE%\.dotnet\tools\.store\docfx\2.76.0\docfx\2.76.0`)
3. Following 2 .nupkg files are existing:
    - docfx.2.76.0.nupkg
    - docfx.nupkg

The same happens with local tools installed with  `<PackageDownload>` and when someone used `dotnet tool restore`. Here you have to check the path  `%USERPROFILE%\.nuget\packages\docfx\2.76.0`

### Expected Behavior

Because of this the following tool path is not resolved

```
    [NuGetPackage(
        packageId: "docfx",
        packageExecutable: "docfx.dll",
        Framework = "net8.0",
        Version = "2.76.0")]
    Tool DocFx 
        => TryGetValue(() => DocFx)
        ?? throw new InvalidOperationException($"{nameof(DocFx)} tool is not assigned. Add <PackageDownload Include=\"{DocFxToolPackageId}\" Version=\"[{DocFxToolPackageVersion}]\" /> to the build project file.");
```

### Actual Behavior

The path should be resolved to `%USERPROFILE%\.nuget\packages\docfx\2.76.0`

### Known Workarounds

- Do not use .NET SDK 8.0
- Do not use `dotnet tool restore`
- Do like the `dotnet-docker` team, and [delete the duplicate file after restoring](https://github.com/dotnet/dotnet-docker/blob/main/src/sdk/8.0/bookworm-slim/amd64/Dockerfile#L47-L48)

### Could you help with a pull-request?

Yes

**Comment by matkoch** [comment]
I'll use #1344 to comment on that.

**Comment by ChristophLindemann** [comment]
@matkoch can we reopen this issue? as it still exists.

It is not a duplicate of #1344. I that issue talk about `tool-manifest` which is not related to this issue.

In the description I linked to the .NET SDK issue and commit, which is the reason for the problem.

**Comment by matkoch** [comment]
I'm not seeing how this issue is different from the other.

1. You are talking about `dotnet tool install -g` which installs to `.dotnet\tools\.store`. This is irrelevant to `NuGetPackageResolver`. [Already pointed out here](https://github.com/nuke-build/nuke/issues/1344#issuecomment-2207358605).
2. You are referring to `dotnet tool install`, but that command relates to tool manifests. So how is it not related?
3. Finally, what exactly isn't covered in your opinion by the [applied change](https://github.com/nuke-build/nuke/commit/e638e6ac9556a678da237eb909b38c22d155abee)?

**Comment by ChristophLindemann** [comment]
hmm, I will have to double check, what is going on. I did not notice that change.

For some reason I still see it. I have to step through on the build server.

**Comment by ChristophLindemann** [comment]
@matkoch it was totally my fault that it still failed. I build the wrong commit on the server. Sorry about that.

I can confirm, that the issue is solved in 8.1.2
